experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -708,42 +712,156 @@ function getMovedMark<T>(
 	return undefined;
 }
 
-function handleLineage<T>(
-	rebasedMark: Mark<T>,
-	lineageRecipients: Mark<T>[],
+type CellBlockList = CellBlock[];
+
+interface CellBlock {
+	readonly cellId: CellId | undefined;
+
+	// The index of the oldest revision where this cell is full in its output context.
+	// May be -Infinity if the cell has been full for all revisions in the base changeset,
+	// or +Infinity if the cell has been empty for all revisions in the base changeset.
+	readonly firstAttachedRevisionIndex: number;
+
+	// The index of the newest revision where this cell is full in its output context.
```

Comment:
Would be nice to have comments about infinity values as is the case for `firstAttachedRevisionIndex`